ContactList/ContactList/Search.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Search.swift
+//  ContactList
+//
+//  Created by Maksym Humeniuk on 5/16/19.
+//  Copyright © 2019 Maksym Humeniuk. All rights reserved.
+//
+
+import Foundation
+
+class Search {
+    
+    static func getFilterResults(from array: [Person], by searchText: String) -> ([Person], [NSAttributedString]) {
+        var resultStringsArray = [NSAttributedString]()
+        
+        let filteredPersons = array.filter({ (person) -> Bool in
+            let sourceString: String
+            
+            if isContains(searchText: searchText, in: person.firstName!) || isContains(searchText: searchText, in: person.lastName!){
+                sourceString = getFullNameString(from: person)
+            } else if isContains(searchText: searchText, in: person.phoneNumber!) {
+                sourceString = person.phoneNumber!
+            } else if isContains(searchText: searchText, in: person.email!){
+                sourceString = person.email!
+            } else {
+                return false
+            }
+            
+            resultStringsArray.append(getResultString(sourceString: sourceString, searchText: searchText))
+            return true
+        })
+        return (filteredPersons, resultStringsArray)
+    }
+    
+    static func getPersonsArrayFromDictionary(from dictionary: [String : [Person]]) -> [Person] {
+        var resultArray = [Person]()
+        for item in dictionary {
+            resultArray += item.value
+        }
+        return resultArray
+    }
+    
+    static func isPersonsFullNameFirstCharCompare(firstPerson: Person, secondPerson: Person) -> Bool {
+        return getFullNameString(from: firstPerson).first == getFullNameString(from: secondPerson).first
+    }
+    
+    static func getFullNameString(from person: Person) -> String {
```

Comment:
I'd recommend to change it to
```
private extension Person {
    var fullName: String {
        return firstName! + (firstName!.isEmpty ? "" : " ") + lastName!
    }
}
```

Are you safe for force unwrapping btw? Both fields are optional and may be `nil`.

Reply:
Don't see any changes here